crates/turbopack-core/src/issue/mod.rs

```diff
@@ -73,6 +73,77 @@ impl Display for IssueSeverity {
     }
 }
 
+/// Represents a section of structured styled text. This can be interpreted and
+/// rendered by various UIs as appropriate, e.g. HTML for display on the web,
+/// ANSI sequences in TTYs.
+#[derive(Clone, Debug)]
+#[turbo_tasks::value(shared)]
+pub enum StyledString {
+    /// Multiple [StyledString]s concatenated into a single line. Each item is
+    /// considered as inline element. Items might contain line breaks, which
+    /// would be considered as soft line breaks.
+    Line(Vec<StyledString>),
+    /// Multiple [StyledString]s stacked vertically. They are considered as
+    /// block elements, just like the top level [StyledString].
+    Stack(Vec<StyledString>),
+    /// Some prose text.
+    Text(String),
+    /// Code snippet.
+    // TODO add language to support syntax hightlighting
+    Code(String),
+    /// Some important text.
+    Strong(String),
+}
+
+impl DeterministicHash for StyledString {
+    fn deterministic_hash<H: turbo_tasks_hash::DeterministicHasher>(&self, state: &mut H) {
+        match self {
+            StyledString::Line(parts) => {
+                "line_start".deterministic_hash(state);
+                for part in parts {
+                    part.deterministic_hash(state);
+                }
+                "line_end".deterministic_hash(state);
```

Comment:
I think we don't need this, as we are just hashing
```suggestion

```

Reply:
These are re: https://github.com/vercel/turbo/pull/6388#discussion_r1385897337

Reply:
I mean `"line_start".deterministic_hash(state);` is enough for hash functions.